content/concepts/module-resolution.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Module Resolution
+contributors:
+    - pksjce
+---
+
+### Introduction
+
+A resolver is a library which helps find the absolute path of a module.
+A module can be required as a dependency from another module as 
+
+```js
+require('/path/to/module')
+//or 
+import mymodule from '/path/to/module'
+```
+
+The dependency module can be from the application code or a third party library. The resolver helps 
+`webpack` find the module code that needs to be included in the bundle for every such `require()/import` statement.
+`webpack` uses [enhanced-resolve](https://github.com/webpack/enhanced-resolve) to resolve file paths while bundling modules.
+
+### Resolving rules in webpack
+
+`webpack` resolves three kinds of file paths
+
+* Absolute paths
+
+```js
+require("/home/me/file");
+require("C:\\Home\\me\\file");
+```
+
+Since we already have the absolute path to the file, no further resolution is required.
+
+* Relative paths
+
+```js
+require("../src/file");
+require("./file");
+```
+
+In this case, the directory of the resource file is taken to be the context directory. The given relative path is joined to the context path to produce the absolute path to the file.
+
+* Module path
+
+```js
+require("module");
+require("module/lib/file");
+```
+
+Modules are searched for inside `moduleDirectories` which are specified using `resolve.moduleDirectories`. These module directories can be under multiple paths if the options under `resolve.root` is an array.
+If the library is not found inside any of these, the resolver defaults to the directories specified in `resolve.fallback`
+Aliasing, ie setting `resolve.alias` to an existing module path, allows you to replace the module path with an alias name during `require/import`.
+
+Once the path is resolved based on the above rule, the resolver checks if the path points to a file or to a directory. If the path points to a file then it is bundled straightaway.
+But if the path points to a folder, then the following steps are taken to find the right file with the right extension.
+* The right file is determined using the `main: "<filename>.js"` field in `package.json`. 
+* If there is no package.json or the main file is not found, `resolve.mainFiles` configuration option is looked up.
+* If this also fails, then it looks for a file named `index` by default.
+* `resolve.extensions` tells the resolver which extensions (eg - `.js, .jsx`) are acceptable for resolution.
+
+### Resolving Loaders
+
+This follows the same rules as the file resolver. But `resolveLoader` configuration can be used to have separate resolution rules for loaders.
+
+### Caching
```

Comment:
Add a newline after each header. Looks more consistent that way. The same would work with paragraphs.